src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if ((this[i] === undefined && i === this.length - 1)
+      || (this[i] === null && i === this.length - 1)) {
+        result += '';
+      } else if (this[i] === undefined || this[i] === null) {
+        result += '' + (separator + '');
+      } else if (i === this.length - 1) {
+        result += this[i];
+      } else {
+        result += this[i] + (separator + '');
```

Comment:
And here you don't need to add an empty string too.

Reply:
My function doesn't work in right way when null is passed as argument. 
[0, 1, 2, 3].join(null)
Expected: "0null1null2null3"
Received: "0123"
So I decided to solve this problem with (separator + '')

Reply:
If I don't know how to solve this issue with null I need to go to chat and ask there or I can ask question about that here?)) How to ask correctly?

Reply:
I got it!  
I need  (separator + '') in order to turn null into string
0 + null = 0  => number + number = number => "0123"
0 + (null + '') = 0null => number + (number + string) = number + string = string => "0null1null2null3"